pkgs/os-specific/darwin/apple-sdk/frameworks.nix

```diff
@@ -46,7 +46,7 @@ with frameworks; with libs; {
   ExceptionHandling       = [];
   FWAUserLib              = [];
   ForceFeedback           = [ CF IOKit ];
-  Foundation              = [ CF libobjc Security ApplicationServices SystemConfiguration ];
+  Foundation              = [ cf-private CF libobjc Security ApplicationServices SystemConfiguration ];
```

Comment:
It's been a while since I looked at this but I'm a little concerned at what this does. This may end up working, but I think it's a little hacky. "ApplicationServices" will already add cf-private as a framework and I think the reason this works is because clang will just use the first "CF" it finds (in this case cf-private). I think we need to look at just including cf-private in CF as that distinction doesn't really exist outside Nixpkgs.

/cc @copumpkin

Reply:
If you're going to add it, please comment why it was needed. In a slightly more ideal world, `cf-private` wouldn't really exist at all, so I want to keep track of what requires it and why.

Reply:
I added a comment to link to this PR because in here it explains why it is needed for ldc and a further reference to the other bug exists anyway.